Sources/NIOCore/AsyncChannel/AsyncChannelInboundStream.swift

```diff
@@ -40,23 +40,25 @@ public struct NIOAsyncChannelInboundStream<Inbound: Sendable>: Sendable {
         }
 
         /// Finished the inbound stream.
+        ///
+        /// - Parameter error: The error to throw, or nil, to finish normally.
         @inlinable
-        public func finish() {
-            self.continuation.finish()
+        public func finish(throwing error: Error?) {
```

Comment:
I think we should default this to `nil`. Having to spell out `finish(throwing: nil)` for a 'happy-path' finish is a bit awkward IMO.